core/src/main/java/com/linecorp/armeria/client/logging/LoggingClientBuilder.java

```diff
@@ -55,4 +60,28 @@
     newDecorator() {
         return this::build;
     }
+
+    /**
+     * Sets the {@link Sampler} that determines which request needs logging.
+     */
+    public LoggingClientBuilder sampler(Sampler<? super ClientRequestContext> sampler) {
+        this.sampler = requireNonNull(sampler, "sampler");
+        return this;
+    }
+
+    /**
+     * Returns the {@link Sampler} that determines which request needs logging.
+     */
+    protected Sampler<? super ClientRequestContext> sampler() {
+        return sampler;
+    }
+
+    /**
+     * Sets the rate at which to sample requests to log. Any number between {@code 0.0} and {@code 1.0} will
+     * cause a random sample of the requests to be logged.
+     */
+    public LoggingClientBuilder samplingRate(float samplingRate) {
```

Comment:
Could you move the new methods (`sampler()` and `samplingRate()` before `build()` and `newDecorator()`?